src/main/java/hudson/scm/SubversionWorkspaceSelector.java

```diff
@@ -102,9 +103,9 @@ public Collection getEnabledFactories(File path, Collection factories, boolean w
     public static volatile int workspaceFormat = SVNAdminArea14.WC_FORMAT;
 
     public static void syncWorkspaceFormatFromMaster() {
-        Hudson h = Hudson.getInstance();
-        if (h!=null)
-            workspaceFormat = h.getDescriptorByType(SubversionSCM.DescriptorImpl.class).getWorkspaceFormat();
+        Jenkins j = Jenkins.getInstanceOrNull();
```

Comment:
This too, perhaps. Really this code is a legacy design that caused problems from SECURITY-144 and needs to be redesigned.

Reply:
Are we sure this code path isn’t visited by the agent jvm?

Reply:
Ditto. It _is_ called from the agent side AFAICT, which is why I am using `getInstanceOrNull`.

Reply:
So you are going to try and load Servlet Spec classes on the agent JVM... that's not good.

You cannot invoke any `Jenkins` method safely from an agent

Reply:
No, you cannot, which is why this code should be rewritten. But I am leaving that for another day.

Reply:
To clarify: `Jenkins.getInstanceOrNull` will normally work OK from an agent (returning null). It _could_ produce Servlet-related linkage errors in relatively obscure environments. But if that was an issue, it has been an issue in this plugin for years. I am just updating to make sure that we do not run into problems with a version of Jenkins core that enforces the documented behavior of `getInstance`.

Reply:
Agreed: [JENKINS-48543](https://issues.jenkins-ci.org/browse/JENKINS-48543)